src/Routing/EdgeDynamicRouting.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_edge\Routing;
+
+use Symfony\Component\Routing\Route;
+use Symfony\Component\Routing\RouteCollection;
+
+/**
+ * Dynamically built routes.
+ */
+class EdgeDynamicRouting {
+
+  /**
+   * Build routes dynamically.
+   */
+  public function routes() {
+    $route_collection = new RouteCollection();
+    $route = new Route('/admin/config/apigee-edge/app/{developer_app_by_name}/credentials',
```

Comment:
Why should is an admin path?

Reply:
Changed it and got rid of the class. Instead I put the route into the route file.